### PR TITLE
feat(config): Re-introduce toString() methods for configuration

### PR DESCRIPTION
**Pull request by jonathanlukas**
Before the refactoring of the client configuration, the whole config was printed on startup.

This PR re-introduces this to verify whether all configuration has been applied correctly.

### CODE REVIEW

File: spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfiguration.java [hunk 1]
```diff
@@ -200,4 +200,8 @@ public boolean useDefaultRetryPolicy() {
     return properties.useDefaultRetryPolicy();
   }
 
+  @Override
+  public String toString() {
+    return "ZeebeClientConfiguration{" + "properties=" + properties + ", commonConfigurationProperties=" + commonConfigurationProperties + ", authentication=" + authentication + ", jsonMapper=" + jsonMapper + ", interceptors=" + interceptors + ", zeebeClientExecutorService=" + zeebeClientExecutorService + '}';
```
Comment (sbuettner) [inline spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfiguration.java:205]: Could this potentially leak credentials?

Comment (jonathanlukas) [reply spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfiguration.java]: True. Set all to hidden as far as I could identify them, see latest commit

File: spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java [hunk 2]
```diff
@@ -305,20 +332,42 @@ public boolean equals(Object o) {
...
+  public static class Cloud {
 
     @Override
     public String toString() {
-      return "Broker{"
-          + "gatewayAddress='"
-          + gatewayAddress
+      return "Cloud{"
+          + "clusterId='"
+          + "***"
```
Comment (sbuettner) [inline spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java:343]: Cluster ID could be useful wdyt?

Comment (sbuettner) [inline spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/properties/ZeebeClientConfigurationProperties.java:343]:
```suggestion
          + "clusterId='"
          + clusterId
```

**Review by sbuettner — APPROVED** [review]

**Pull request merged**